cpp-fs/storage/storage/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+project(storage CXX)
+
+set(STORAGE_NAME storagelib)
+add_library(${STORAGE_NAME} storage.cpp)
+target_link_libraries(${STORAGE_NAME} PRIVATE Boost::headers tl::expected)
```

Comment:
as far as I can see `expected` is a part of the storage interface, so it should be linked here using `PUBLIC` keyword.